.github/workflows/bindings_python_ci.yml

```diff
@@ -78,6 +78,6 @@ jobs:
         shell: bash
         run: |
           set -e
-          pip install dist/pyiceberg_core-*.whl --force-reinstall
-          pip install pytest
-          pytest -v
+          pip install hatch==1.12.0
```

Comment:
How about split workflow related changes to another PR?

Reply:
Agreed. I put up this PR as a follow up: https://github.com/apache/iceberg-rust/pull/572